### PR TITLE
Add/Remove Content for Revised Git-Sync Architecture

### LINKED ISSUE #1490: Software: New Git-Sync Architecture
**Request type:**

- [ ] **New document:** Add the `docs-content` label to this issue.
- [x] **Addition to an existing document**: Add the `enhancement` label to this issue.

**What is the purpose of the document/enhancement? Select all that apply:**

- [ ] Document a new feature for an upcoming release. Add the appropriate release version label to this issue.
- [ ] Information for an existing feature that isn't yet documented.
- [x] Additional considerations/ reference material for an existing document.

**Problem/feature that requires documentation:**
We have implemented a better architecture for git-sync for 0.31. We need to update this [doc](https://docs.astronomer.io/software/deploy-git-sync) and remove the experimental feature. We can also include a quick blurb about the new architecture if that makes sense. I don't know if there are configuration changes.

Main issues:
- https://github.com/astronomer/issues/issues/4059
- https://github.com/astronomer/issues/issues/3980

@danielhoherd and @bote795 are the subject matter experts.

**Are you the right person to review a draft of the document for technical accuracy? If not, who is?**

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #1490

### CODE REVIEW

**Comment by danielhoherd** [comment]
@kwcanuck here's a doc i wrote up about the feature https://github.com/astronomer/airflow-chart/blob/master/docs/git-sync-relay.md

**Comment by kwcanuck** [comment]
@cmarteepants or @danielhoherd should the information provided in the [overview doc](https://github.com/astronomer/airflow-chart/blob/master/docs/git-sync-relay.md) be incorporated into the existing topic, or is this for Astronomer use only? Is this a process we want users to use?

**Comment by kwcanuck** [comment]
@bote795 indicated that "....the current documentation stays the same we just changed the implementation causing it to be better for our customers". @danielhoherd recommended that the interval time be updated.

I have made the recommended changes and also a few changes to match our current standards.

**Review by bote795 — APPROVED** [review]

File: software/deploy-git-sync.md [hunk 1]
```diff
@@ -54,13 +50,13 @@ Workspace editors can configure a new or existing Airflow Deployment to use a gi
 
     - **Repository URL**: The URL for the Git repository that hosts your Astro project
     - **Branch Name**: The name of the Git branch that you want to sync with your Deployment
-    - **Sync Interval**: The time interval between checks for updates in your Git repository, in seconds. A sync is only performed when an update is detected. We recommend a 1 second interval
+    - **Sync Interval**: The time interval between checks for updates in your Git repository, in seconds. A sync is only performed when an update is detected. Astronomer recommends a minimum  interval of 60 seconds.
```
Comment (bote795) [inline software/deploy-git-sync.md:53]: idk what the recommended value should be that is the only thing

Comment (kwcanuck) [reply software/deploy-git-sync.md:53]: @danielhoherd recommended this change.

Comment (danielhoherd) [reply software/deploy-git-sync.md:53]: git-sync (the tool) has a default of 10s. Airflow's default is 60s. I think that is a reasonable number. 1s is unreasonable.

**Comment by jwitz** [comment]
Looks good, thank you Rob! Keeping this in draft until 0.31 is out.

File: software/deploy-git-sync.md (hunk 1 above)
Comment (danielhoherd) [reply software/deploy-git-sync.md:53]: FWIW i opened 2 issues to address the sync times
- https://github.com/astronomer/issues/issues/5216
- https://github.com/astronomer/issues/issues/5213

**Review by ajayy004 — APPROVED** [review]

File: software/deploy-git-sync.md [hunk 2]
```diff
@@ -5,14 +5,10 @@ id: deploy-git-sync
 description: Push DAGs to your Airflow Deployment on Astronomer Software using git-sync.
 ---
 
-Starting with Astronomer v0.27, you can deploy DAGs to an Astronomer Deployment using [git-sync](https://github.com/kubernetes/git-sync). After setting up this feature, you can deploy DAGs from a Git repository without any additional CI/CD. DAGs deployed via git-sync automatically appear in the Airflow UI without requiring additional action or causing downtime.
+Starting with Astronomer v0.27, you can deploy DAGs to an Astronomer Deployment using [git-sync](https://github.com/kubernetes/git-sync). After setting up this feature, you can deploy DAGs from a Git repository without any additional CI/CD. DAGs deployed with git-sync automatically appear in the Airflow UI without requiring additional action or causing downtime.
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deploy-git-sync.md:8]: @danielhoherd thanks for the feedback. Unfortunately, the use of _via_ is not recommended  in technical documentation. Here is the entry from our selected style guide:

image

[the remark this answers is not in the document]
Comment (danielhoherd) [reply software/deploy-git-sync.md:8]: Why?

[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deploy-git-sync.md:8]: It's a Latin term and we avoid using Latin (etc., eg,) in our documentation. Latin terms are not easily understood by all audiences. If you're curious about technical writing standards, see the _[Google developer documentation style guide](https://developers.google.com/style)_ (our primary style guide refence) or the _[Microsoft Style Guide](https://learn.microsoft.com/en-us/style-guide/welcome/)_ (our secondary reference). You might also be interested in reviewing the _[Astronomer Style Guide](https://www.notion.so/astronomerio/Astronomer-Style-Guide-5d2389d462734e149d2e66a5cf606228)_.

**Pull request merged**